sw/device/tests/chip_power_idle_load.c

```diff
@@ -0,0 +1,516 @@
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_isrs.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#include "alert_handler_regs.h"  // Generated.
+#include "aon_timer_regs.h"
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"  // Top-specific Definitions
+#include "pwm_regs.h"
+
+//==============================================================================
+// Global Variables
+
+typedef void (*isr_handler)(void);
+static volatile isr_handler expected_isr_handler;
+static volatile dif_rv_core_ibex_nmi_state_t nmi_state;
+static volatile bool nmi_fired = false;
+static volatile bool ext_irq_fired = false;
+static volatile bool irq_is_pending = false;
+
+static dif_aon_timer_t aon_timer;
+static dif_rv_core_ibex_t rv_core_ibex;
+static dif_pwrmgr_t pwrmgr;
+static dif_rv_timer_t rv_timer;
+static dif_alert_handler_t alert_handler;
+static dif_pwm_t pwm;
+static dif_pinmux_t pinmux;
+static dif_otp_ctrl_t otp_ctrl;
+static dif_gpio_t gpio;
+
+//==============================================================================
+OTTF_DEFINE_TEST_CONFIG();
+
+//==============================================================================
+// Exception handlers
+// ...using the default OTTF exception handlers...
+
+//==============================================================================
+// ISRs
+
+// ------------------------------------
+// OTTF External-IRQ handler
+// (this functions overrides the OTTF weak definition)
+void ottf_external_isr(void) {
+  LOG_INFO("SW_INFO: got external IRQ");
+  ext_irq_fired = true;
+}
+
+// ------------------------------------
+// OTTF external NMI internal IRQ handler.
+// This functions overrides the OTTF weak definition.
+// It calls `expected_isr_handler` function pointer that is previously set by
+// the test for the specific peripheral.
+void ottf_external_nmi_handler(void) {
+  nmi_fired = true;
+
+  expected_isr_handler();  // this calls the specific handler as defined in
+                           // "test_main"
+
+  CHECK_DIF_OK(dif_rv_core_ibex_get_nmi_state(
+      &rv_core_ibex, (dif_rv_core_ibex_nmi_state_t
+                          *)&nmi_state));  // get the state before clearing
+  CHECK_DIF_OK(dif_rv_core_ibex_clear_nmi_state(
+      &rv_core_ibex, kDifRvCoreIbexNmiSourceAll));  // clear
+}
+
+// ------------------------------------
+// Specific handlee for the wdog bark NMI IRQ.
+static void wdog_irq_handler(void) {
+  bool is_pending;
+
+  // The watchdog bark external interrupt is also connected to the NMI input of
+  // rv_core_ibex. We therefore expect the interrupt to be pending on the
+  // peripheral side (the check is done later in the test function).
+  CHECK_DIF_OK(dif_aon_timer_irq_is_pending(
+      &aon_timer, kDifAonTimerIrqWdogTimerBark,
+      &is_pending));  // (this wants "bool" not "volatile bool")
+  irq_is_pending = is_pending;
+
+  // Stop the watchdog.
```

Comment:
Many of these comments are not needed because the function name and parameters are self explanatory. We use the OpenTitan C/C++ style guide is based on the Google one. Here are more guidelines on what we cover as part of the style review.

https://google.github.io/styleguide/cppguide.html#Implementation_Comment_Donts

Reply:
Done.